### ISSUE TITLE
No Color Temperature Support for BNETA GU10 RGBW Bulb

### ISSUE THREAD

**Issue by asifkassam** [tuya/tuya-home-assistant#131]
**Describe the bug**
No Temperature support for BNETA GU10 RGBW Bulbs

**Expected behavior**
Expect to be able to control white color temperature as is the case 

**Screenshots**
![image](https://user-images.githubusercontent.com/67918736/125193312-17d29280-e24c-11eb-9e2e-711b0d78a3a5.png)

**Home Assistant Version**
- e.g. 2021.7.1

**Device info (please complete the following information, which can be found in [log](https://github.com/tuya/tuya-home-assistant/wiki/How-to-get-the-log)):**
like this:
      "product_id": "eyjmbtxjcw2ghx2m",
      "product_name": "BNETA GU10 RGBW",
      "status": [
        {
          "code": "switch_led",
          "value": false
        },
        {
          "code": "work_mode",
          "value": "white"
        },
        {
          "code": "bright_value_v2",
          "value": 516
        },
        {
          "code": "colour_data_v2",
          "value": "{\"h\":360,\"s\":1000,\"v\":1000}"
        },
        {
          "code": "scene_data_v2",
          "value": "{\"scene_num\":3,\"scene_units\":[{\"bright\":1000,\"h\":0,\"s\":0,\"temperature\":1000,\"unit_change_mode\":\"static\",\"unit_gradient_duration\":13,\"unit_switch_duration\":14,\"v\":0}]}"
        },
        {
          "code": "countdown_1",
          "value": 0
        },
        {
          "code": "control_data",
          "value": ""
        }

**Device specifications (please complete the following information, which can be found in [log](https://github.com/tuya/tuya-home-assistant/wiki/How-to-get-the-log)):**
      "product_id": "eyjmbtxjcw2ghx2m",
      "product_name": "BNETA GU10 RGBW",
      "status": [
        {
          "code": "switch_led",
          "value": false
        },
        {
          "code": "work_mode",
          "value": "white"
        },
        {
          "code": "bright_value_v2",
          "value": 516
        },
        {
          "code": "colour_data_v2",
          "value": "{\"h\":360,\"s\":1000,\"v\":1000}"
        },
        {
          "code": "scene_data_v2",
          "value": "{\"scene_num\":3,\"scene_units\":[{\"bright\":1000,\"h\":0,\"s\":0,\"temperature\":1000,\"unit_change_mode\":\"static\",\"unit_gradient_duration\":13,\"unit_switch_duration\":14,\"v\":0}]}"
        },
        {
          "code": "countdown_1",
          "value": 0
        },
        {
          "code": "control_data",
          "value": ""
        }

**Additional context**
Add any other context or logs about the problem here.

**Comment by tsutsuku** [comment]
@asifkassam  It's an RGBW bulb. It does not support temperature.

**Comment by asifkassam** [comment]
you're right, my bad on this one, it is the same in the app, only difference is that the HA RGB color wheel can give you a cool white  and there is then no way to go back to the bulbs normal warm white color, whereas in the app, reverting to white  works, with no CT slider and is always warm white

**Comment by xcraftman52** [comment]
I am having the same issue

**Comment by iotinkognito** [comment]
Having same issue.

**Comment by mikesellt** [comment]
Same issue here.  I opened a new issue for Merkury bulbs, but it looks like the same issue as this.

**Comment by mikesellt** [comment]
I just tried changing values under "customization" for over an hour.  I could get different brightness sliders and other options to show up, but they wouldn't change anything.  I ended up going back to default, and again, color and brightness work (in color mode).  I can't get any kind of "white" settings (temp, brightness) to work at all.  I checked this against the debug values in iot.tuya.com for the device, but none of those values worked either when trying them in HA.

**Comment by asifkassam** [comment]
perhaps this could help

https://www.home-assistant.io/integrations/light.mqtt/#brightness-and-rgbw-support

**Comment by rmensing** [comment]
I am having the same issue with a similar light. A **Merkury BW904 Smart Bulb**.  I am a sometimes hobby coder so I know enough to read the code and have an idea of what it is doing but I don't know enough right now about HA Dev to fix it myself.  I wanted to add my observations about what is/is not happening and, at least generally, what needs to be done I think.

To start with, in a nutshell, there are few types of lights that fall under category "dj" :
1. White - single temperature
2.  White - adjustable temperature
3. Color only
4. Color with white - single temp 
5. Color with white - adjustable temp

Referencing the code for [lights on the Tuya_v2  github.](https://github.com/tuya/tuya-home-assistant/blob/master/custom_components/tuya_v2/light.py)
From what I see, each time the integration polls the API for the device it looks at the status responses and determines what options the device supports.  Currently, the integration only sends 3 of the things to HA:

(code lines 14-16)
```
    SUPPORT_BRIGHTNESS,
    SUPPORT_COLOR,
    SUPPORT_COLOR_TEMP
```
If your device does not have Color Temp in the status, you do not get color temp adjustment in the UI even if you try and change it in the customizations.  

When you change the brightness, the code will look at the **Work_Mode** to see if it is color or white so it knows how to adjust the brightness but it will not change the work mode.
(Code lines 164 - 183)

If the color wheel on the UI is used to change the color while it is in _white_ **Work_Mode**, it will change the **Work_Mode** to color.
(Code lines 185 - 210, Specifically lines 209-210) 

The ONLY time the **Work_Mode** is changed to _white_  is if the **Color_Temp** is changed and it is not in _white_ **Work_Mode**
(Code lines 212 - 233, Specifically lines 232 - 233)

As it stands right now, the easiest way that I see for them to fix this is to add the color temp option to the lights that have a **work_mode** status.  or at least make it so the color temp option is able to be set in the customizations.  The only potential issue I see is that the **Work_Mode** status does not necessarily mean it supports both color and white because it also supports other options (ref. [Tuya Standard instruction set of lights (dj)](https://developer.tuya.com/en/docs/iot/f?id=K9i5ql3v98hn3#title-0-Standard%20instruction%20set%20of%20lights%20(dj)).

An additional issue I see creating upon us that will have an effect on this and maybe why they aren't pushing code changes heavily is, the way light features/color modes for lights are being handled in Home Assistant is changing some of them have been deprecated and will be removed with Home Assistant Core 2021.10 (Oct. I am assuming since we are on 2021.8.x now)

Please feel free to correct me where I am wrong or just add your input.

Cross posted  to [#131](https://github.com/tuya/tuya-home-assistant/issues/131), [#234](https://github.com/tuya/tuya-home-assistant/issues/234), and [#308](https://github.com/tuya/tuya-home-assistant/issues/308)
